Resolve issues with std::basic_ios and std::basic_stringstream usage

davidlt opened the pull request:
`std::cout` does not directly support `std::basic_stringstream` as
argument, but because `std::basic_stringstream` inherits `std::basic_ios`
and it supports `operator void*` (until C++11) one could have pushed it
to `std::cout`. That is wrong, because `operator void*() const` does not
provide string content of `std::basic_stringstream`. Instead use `str()`
method to get a string copy of `std::basic_stringstream` content.

Instead of using `operator void*() const;` to check if stream is
good/bad. You have to use `explicit operator bool() const` in C++11 or
above or directly use `fail()` method.

Signed-off-by: David Abdurachmanov David.Abdurachmanov@cern.ch

Commit:
Resolve issues with std::basic_ios and std::basic_stringstream usage

`std::cout` does not directly support `std::basic_stringstream` as
argument, but because `std::basic_stringstream` inherits `std::basic_ios`
and it supports `operator void*` (until C++11) one could have pushed it
to `std::cout`. That is wrong, because `operator void*() const` does not
provide string content of `std::basic_stringstream`. Instead use `str()`
method to get a string copy of `std::basic_stringstream` content.

Instead of using `operator void*() const;` to check if stream is
good/bad. You have to use `explicit operator bool() const` in C++11 or
above or directly use `fail()` method.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

cmsbuild commented:
A new Pull Request was created by @davidlt  for CMSSW_7_6_X.

Resolve issues with std::basic_ios and std::basic_stringstream usage

It involves the following packages:

SimCalorimetry/EcalElectronicsEmulation
SimCalorimetry/EcalTrigPrimProducers
SimG4CMS/FP420
SimG4CMS/Forward
SimMuon/CSCDigitizer


@cmsbuild, @civanch, @mulhearn, @mdhildreth can you please review it and eventually sign? Thanks.
@makortel, @jhgoh, @argiro this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

cmsbuild commented:
+1
Tested at: 22339a3d09dcf57f28a1fe950d74d2b3bfc475b0
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11308/8153/summary.html

mulhearn commented:
+1

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11308/8153/summary.html

civanch commented:
@davidlt , in SimCalorimetry/EcalElectronicsEmulation/src/EcalSimRawData.cc there are similar constructions in lines 208, 286, 356 which are not yet changed.

davidlt commented:
@civanch I changed that particular line from `operator!` to `fail()`, because I believe `fail()` provides more details on what is checked. Other than that, both do exactly the same. I can revert that line back if needed for consistency.

civanch commented:
@davidlt , I mean to do the same construction if(f.fail()) in lines 208, 286, 356

Commit:
Move from operator! to fail() for checking stream status

`fail()` and `operator!` work in the same way, but `fail()` provides more
details, i.e. makes code easier to understand.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

cmsbuild commented:
Pull request #11308 was updated. @civanch, @Dr15Jones, @cvuosalo, @emeschi, @ianna, @mdhildreth, @monttj, @cmsbuild, @ggovi, @diguida, @cerminar, @deguio, @slava77, @mommsen, @vadler, @mmusich, @mulhearn, @danduggan can you please check and sign again.

diguida commented:
please test

davidlt commented:
Done, but I had to rebase. Forgot there was other branches in work in the same work dir. Hopefully cms-bot resolves the labels.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #11308 was updated. @cmsbuild, @civanch, @mulhearn, @mdhildreth can you please check and sign again.

cmsbuild commented:
+1
Tested at: bdcaf4206f4f9f428f86641b2a31770c56969348
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11308/8174/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/02bcaa58efb84fd2d9a4d54a5b1d5f3c9faf14c7
https://github.com/cms-sw/cmssw/commit/3fd9c445561d2ddf3e0ca165fc14ef81983365a1
https://github.com/cms-sw/cmssw/commit/0d27261c38fa6df94873e658e9a32e32362a305f
https://github.com/cms-sw/cmssw/commit/79e86f7268afe4534dac5ec29efa320338ccef86
https://github.com/cms-sw/cmssw/commit/414bc08085e4df5877b79ffaa9a0033d3e38c722
https://github.com/cms-sw/cmssw/commit/0de92fb7f16a9691547582adb9696ab0e565a250
https://github.com/cms-sw/cmssw/commit/dce4ad863fb34b5a83bf1701164bc83f92161b9d
https://github.com/cms-sw/cmssw/commit/09d2d6fa45009d37b7be643e96e6000ad8f0c6db
https://github.com/cms-sw/cmssw/commit/2cd4f46dd2bd6733ddae47d80c2df48757ea3c3b
https://github.com/cms-sw/cmssw/commit/32be3168b69f410f9a34cc379a47b276c81bea19
https://github.com/cms-sw/cmssw/commit/fe888316c8e6dd4006d20b733441e95ca91c9242
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11308/8174/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11308/8174/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11308/8174/summary.html

civanch commented:
+1
